src/scope/core/pipelines/interface.py

```diff
@@ -58,73 +57,195 @@ def validate_resolution(cls, v: dict[str, int]) -> dict[str, int]:
     model_config = {"extra": "allow"}
 
 
+class PipelineDefaults(BaseModel):
+    """Mode-aware defaults for pipelines supporting text and video generation modes.
+
+    Pipelines declare their native mode and provide mode-specific configurations.
+    """
+
+    native_generation_mode: str = Field(
+        ...,
+        description="Native generation mode for this pipeline (text or video)",
+    )
+    modes: dict[str, PipelineModeConfig] = Field(
+        ...,
+        description="Mode-specific defaults. Keys are 'text' and 'video'",
+    )
+
+    model_config = {"extra": "forbid"}
+
+
 class Pipeline(ABC):
-    """Abstract base class for all pipelines."""
+    """Abstract base class for all pipelines.
+
+    Note: Use GENERATION_MODE_TEXT and GENERATION_MODE_VIDEO constants from defaults.py
+    when overriding NATIVE_GENERATION_MODE in subclasses.
+
+    Architecture Note - ISP Consideration:
+    All pipelines are currently required to declare generation mode support (text/video)
+    even if they only actively use one mode. This is an acknowledged Interface Segregation
+    Principle deviation, accepted because:
+    1. All pipelines will support both modes in future iterations
+    2. The temporary violation simplifies the architecture during the transition
+    3. The overhead is minimal (declaration only, no forced implementation)
+    """
+
+    NATIVE_GENERATION_MODE: str = "text"
+
+    @classmethod
+    def _create_mode_configs(
```

Comment:
nit: I'd probably move helper functions to a separate file to keep the `interface.py` clean. I think the `interface.py` should be pretty minimalistic, because that is something someone will read if they want to develop their own pipelines.